plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSecurityConfig.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+import static io.trino.plugin.deltalake.DeltaLakeSecurityConfig.DeltaLakeSecurity.ALLOW_ALL;
+import static io.trino.plugin.deltalake.DeltaLakeSecurityConfig.DeltaLakeSecurity.READ_ONLY;
+
+public class TestDeltaLakeSecurityConfig
+{
+    @Test
+    public void testDefaults()
+    {
+        assertRecordedDefaults(recordDefaults(DeltaLakeSecurityConfig.class)
+                .setSecuritySystem(ALLOW_ALL));
```

Comment:
I would prefer SYSTEM as default

Reply:
I based myself on the Iceberg security default:

https://github.com/trinodb/trino/blob/627e395b43937d950185580de49ea80c5527456f/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSecurityConfig.java#L30

Wouldn't we want to keep this consistent with Iceberg?

Reply:
I would change the Iceberg too. From user perspective ALLOW_ALL and and SYSTEM do not differ match, other than ALLOW_ALL pretends that it manages roles (while actaully it has no roles) so it limits system roles use cases.

I think we could even consider removal of ALLOW_ALL.

Hence I would prefer to use SYSTEM here, and then update Iceberg too.